### DISCUSSION TITLE
Help with addressing multiple inverters by esp32 and RS485... Current status of this project including/regarding new production hardware?

### DISCUSSION THREAD

**Discussion by garylankford8** [syssi/esphome-soyosource-gtn-virtual-meter#190] [Q&A]
I just bought two of these inverters on AlliExpress after finding this project... If I had read more deeply, I would have known about the later production "silent" inverters and  probably not bought them, lol!

The two I bought came with wifi dongles and limiters. If I could have found any that came with screens and wifi, I would have bought them; kinda sorry I gave up screen for wifi, but I was thinking home assistant...

I put together an esp32 with 2 RS485-ttl modules using the 'esp32 multiple inverter example' code and get the error (same as if the modbus plug isn't plugged in), obviously no communication.

```[09:24:00][W][soyosource_inverter:113]: 'gtn_vm_l1': The inverter didn't respond to the last 15 status requests```

Is there a path to addressing 'silent' inverters, not explored before?

More about my situation. I have a 9kw micro-inverter and grid-tied ground mount array and net metering. These panels are all arrayed due south and have a fairly open horizon, but do lose some sun late in the day. I also have a Magnum MS4448pae off-grid inverter and 3x SOK 100A 48V batteries that I switch on to at night, but are really intended for emergency/SHTF.

A separate system is 1200W PV charging a 60V 100Ah lead acid (LA) battery set through an Outback Power FLEXMax60, that I use to run my 48V well pump 15min twice a day... These panels can be rotated manually from due south to due west. There is a fair amount of potential power being wasted there...

My main purpose/use for these inverters is to tap that wasted power, making power from those LA batteries available to my grid (gtn battery mode), particularly by extending and increasing power production later in the day. That does not really require any limiter, virtual or physical, since I am grid tied, but I did like the idea of home assistant integration for monitoring battery and inverter parameters. The split phase line feeding the well house already has power monitoring as does my main (Emporia Vue 2 on esphome local custom firmware) which I configured to pos/neg. I also thought the limiter control might help with managing my AC coupled system off-grid later on.

First step was to get this project working... As you know, syssi, I'm not the brightest bulb, but I can follow instructions. Any help?

**Comment by syssi** [comment]
First of all: Which Soyosource model do you own exactly? The one with a display or a WiFi dongle?

**Reply by garylankford8** [reply]
The one with the wifi dongle... GTN1200LIM-W 110-120V

Grid Tie Inverter 1200W with Limiter,WIFI APP Remote Control ,Battery Discharge Mode Solar Panel Grid Tie Micro Inverter
https://a.aliexpress.com/_mq2ggMg

**Reply by syssi** [reply]
Good choice! I will provide some instructions tomorrow. Being `silent` isn't a big issue here.

**Reply by garylankford8** [reply]
Thanks, syssi! No need for further instructions at this time. When you said wifi, the bulb between my ears brightened by a few watts... I was so hung up on the esp32 multiple uart thing that it never occurred to me that the wifi dongle had some kind of wifi chip, and that might be an esp8266...

So, with that bulb alternately dimming and brightening as power ebbed and flowed through my feeble mind, I bumbled my way through a few more posts in discussions and then in issues (where I hadn't even thought to look) and found your "How to flash..."

Given that, I still trialed and errored until I finally got everything right and flashed the wifi-dongle-example to my two dongles.

Limiter will have to wait a week until Amazon gets around to delivering the smaller RS485 boards that will fit into the dongle case... Despite being a prime member, boards ordered last night won't arrive until next Monday out here in the country...
